NineChronicles.Headless.Executable/Commands/GenesisCommand.cs

```diff
@@ -14,125 +11,210 @@
         )
         {
-            var options = new JsonSerializerOptions
+            Console.WriteLine("Processing currency for genesis...");
+            if (config is null)
             {
-                AllowTrailingCommas = true,
-                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
-            };
+                Log.Information("CurrencyConfig not provided. Skip setting...");
+                initialMinter = new PrivateKey();
+                initialDepositList = new List<GoldDistribution>
+                {
+                    new()
+                    {
+                        Address = initialMinter.ToAddress(), AmountPerBlock = DefaultCurrencyValue,
+                        StartBlock = 0, EndBlock = 0
+                    }
+                };
+                return;
+            }
 
-            string json = File.ReadAllText(configPath);
-            GenesisConfig genesisConfig = JsonSerializer.Deserialize<GenesisConfig>(json, options);
-            if (string.IsNullOrEmpty(genesisConfig.TablePath))
+            if (string.IsNullOrEmpty(config.Value.InitialMinter))
             {
-                throw Utils.Error("TablePath is not set.");
+                Log.Information("Private Key not provided. Create random one...");
+                initialMinter = new PrivateKey();
+            }
+            else
+            {
+                initialMinter = new PrivateKey(config.Value.InitialMinter);
             }
 
-            Dictionary<string, string> tableSheets = Lib9cUtils.ImportSheets(genesisConfig.TablePath);
+            initialDepositList = new List<GoldDistribution>();
+            if (config.Value.InitialCurrencyDeposit.Count == 0)
+            {
+                Log.Information("Initial currency deposit list not provided. " +
+                                $"Give initial ${DefaultCurrencyValue} currency to InitialMinter");
+                initialDepositList.Add(new GoldDistribution
+                {
+                    Address = initialMinter.ToAddress(),
+                    AmountPerBlock = DefaultCurrencyValue,
+                    StartBlock = 0,
+                    EndBlock = 0
+                });
+            }
+            else
+            {
+                initialDepositList = config.Value.InitialCurrencyDeposit;
+            }
+        }
 
-            Lib9cUtils.CreateActivationKey(
-                out List<PendingActivationState> pendingActivationStates,
-                out List<ActivationKey> activationKeys,
-                genesisConfig.ActivationKeyCount);
+        private void ProcessAdmin(AdminConfig? config, PrivateKey initialMinter, out AdminState adminState)
+        {
+            Console.WriteLine("Processing admin for genesis...");
+            // FIXME: If the `adminState` is not required inside `MineGenesisBlock`,
+            //        this logic will be much lighter.
+            adminState = new AdminState(new PrivateKey().ToAddress(), 0);
```

Comment:
There is a `default` keyword. Then it may create an admin state like `new AdminState(new Address("0x0000...000", 0)`.